src/condor_utils/manifest.cpp

```diff
@@ -0,0 +1,153 @@
+/***************************************************************
+ *
+ * Copyright (C) 2022, Condor Team, Computer Sciences Department,
+ * University of Wisconsin-Madison, WI.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+#include <string.h>
+#include <string>
+#include <map>
+#include <fstream>
+
+#include <openssl/hmac.h>
+#include <openssl/sha.h>
+#include "ssl_version_hack.h"
+
+#include "manifest.h"
+#include "checksum.h"
+#include "stl_string_utils.h"
+#include "AWSv4-impl.h"
+
+
+namespace manifest {
+
+int
+getNumberFromFileName( const std::string & fn ) {
+	const char * fileName = fn.c_str();
+	if( fileName == strstr( fileName, "MANIFEST." ) ) {
+		char * endptr;
+		const char * suffix = fileName + 9;
+		if( *suffix != '\0' && isdigit(*suffix) ) {
+			int manifestNumber = strtol( suffix, & endptr, 10 );
+			if( *endptr == '\0' ) {
+				return manifestNumber;
+			}
+		}
+	}
+	return -1;
+}
+
+bool
+validateFile( const std::string & fileName ) {
```

Comment:
I didn't think it was necessary to name the function `manifest::validateManifestFile()`, but I suppose if you didn't find the current name clear, I should change it.  I'll add the comment.

Reply:
My first thought when seeing the function in the header was that it would check that a file matched its checksum in the manifest. Then I realized there should be more arguments for that. I think a comment stating what file is being validated is needed.